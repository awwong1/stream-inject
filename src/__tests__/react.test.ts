// tslint:disable: no-implicit-dependencies
import { createReadStream } from "fs";
import { join } from "path";
import { createElement } from "react";
import { renderToNodeStream } from "react-dom/server";
import { StreamInjector } from "..";
import { StubComponent } from "./stub";

describe("combination", () => {
  it("should handle a basic react ssr workflow", done => {
    expect.assertions(1);
    const htmlPath = join(__dirname, "test.html");
    const htmlStream = createReadStream(htmlPath, { encoding: "utf8" });
    const reactStream = renderToNodeStream(createElement(StubComponent));

    const testStream = htmlStream
      .pipe(new StreamInjector("{body}", reactStream))
      .pipe(new StreamInjector("{reduxStateCode}", (
        "<script>window.__PRELOADED_REDUX_DATA__= " +
        JSON.stringify({ a: 1 }).replace(/</g, "\\u003c") +
        "</script>"
      )));

    const holder: string[] = [];
    testStream.on("data", (chunk: Buffer) => holder.push(chunk.toString()));
    testStream.on("end", () => {
      expect(holder.join("")).toMatchSnapshot();
      done();
    });
  });
});
