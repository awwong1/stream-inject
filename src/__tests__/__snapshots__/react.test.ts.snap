// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`combination should handle a basic react ssr workflow 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">

<head>
  <meta charset=\\"utf-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\">
  <meta name=\\"author\\" content=\\"Transform Stream Use Case\\">
</head>

<body>
  <div id=\\"root\\"><div data-reactroot=\\"\\">Stub Test Component</div></div>
</body>
<script>window.__PRELOADED_REDUX_DATA__= {\\"a\\":1}</script>

</html>
"
`;
